components/supervisor/frontend/src/ide/ide-metrics-service-client.ts

```diff
@@ -70,6 +70,7 @@ export class IDEMetricsServiceClient {
                 method: "POST",
                 body: JSON.stringify(params),
                 credentials: "omit",
+                priority: "low",
```

Comment:
@filiptronicek are you able to create the scenario where you can see metrics stall / get stuck in Pending?

Reply:
nope, and not even with forced HTTP/1.1. Theoretically, though, if two requests were made at the exact same time in the dashboard now (for instance, one for starting a workspace or watching its status), that one would get prioritized, making us less prone to issues with the 6 concurrent HTTP/1.1 TCP connections per origin.

Reply:
Will priority work with HTTP/1/1? I see references to HTTP/2 and HTTP/3 [here](https://imkev.dev/fetchpriority-opportunity#fetchpriority-all-the-things).

It looks like it is broadly supported, [except in Firefox](https://developer.mozilla.org/en-US/docs/Web/API/HTMLScriptElement/fetchPriority).

Can GPT drum up a simple set of code to help test whether this works? For example, to make an artificially high number of calls in two different groups (a high and a low)?

Reply:
> Will priority work with HTTP/1/1? I see references to HTTP/2 and HTTP/3 [here](https://imkev.dev/fetchpriority-opportunity#fetchpriority-all-the-things).

I don't believe this to be dependent on the HTTP version. The linked article just refers to h2 and h3 in relation to the data from HTTP Archive. 

> It looks like it is broadly supported, [except in Firefox](https://developer.mozilla.org/en-US/docs/Web/API/HTMLScriptElement/fetchPriority).

Yes, Firefox thankfully just ignores it, so there's no need for any special handling.

> Can GPT drum up a simple set of code to help test whether this works? For example, to make an artificially high number of calls in two different groups (a high and a low)?

I'll take a look.

Reply:
I have gotten some pretty ok results when firing 500 requests at once with [this fiddle](https://jsfiddle.net/hyL8pefk/):
```
#1
"Mean response time for priority high: 2001.64 ms"
"Mean response time for priority low: 2128.59 ms"

#2
"Mean response time for priority high: 2001.64 ms"
"Mean response time for priority low: 2128.59 ms"
```

Reply:
I tested while on HTTP/1.1, the low priority group always takes 500ms longer. 👍 

[Here's an updated fiddle](https://jsfiddle.net/7s59mzrg/) that hits an endpoint and introduces some delay. It's really slow with HTTP/1.1. But, it shows high priority is "much better".

Results for updated fiddle:
☁️ "Running fiddle"
"Priority high: Mean response time: 43465.43 ms, Count: 46"
"Priority low: Mean response time: 125647.33 ms, Count: 54"

Reply:
I think we're not going to miss out on them, rather just let them queue to a later time. I'm happy to take a look when this hits production